File: QA/tests/test_about_redesign.py
import unittest
from pathlib import Path


class AboutRedesignTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.root = Path(__file__).resolve().parents[2]
        cls.about_html = (cls.root / "about/index.html").read_text(encoding="utf-8")
        cls.about_css = (cls.root / "styles/about.css").read_text(encoding="utf-8")

    def test_opening_copy_exists(self):
        for phrase in [
            "OPERATING BACKGROUND",
            "Execution is the Strategy",
            "The work is rarely blocked by strategy alone.",
            "The objective is not more process. It is a clearer, more accountable way for work to move from intent to outcome.",
        ]:
            self.assertIn(phrase, self.about_html)

    def test_five_chapter_structure_exists(self):
        self.assertEqual(self.about_html.count('class="era-header"'), 5)
        self.assertIn('id="enterprise-scale"', self.about_html)
        self.assertIn('id="global-delivery-leadership"', self.about_html)
        self.assertIn('id="global-infrastructure-advisory"', self.about_html)
        self.assertIn('id="ai-enabled-operating-systems"', self.about_html)
        self.assertIn('id="fractional-integration-leadership"', self.about_html)

    def test_old_philosophy_card_removed(self):
        self.assertNotIn('class="philosophy-stack"', self.about_html)
        self.assertNotIn('Systems Over Heroics', self.about_html)
        self.assertIn("The strongest operating systems make execution visible.", self.about_html)

    def test_supporting_css_exists(self):
        self.assertIn(".about-chapter-nav", self.about_css)
        self.assertIn(".arc-item", self.about_css)
        self.assertIn("border: 1px solid var(--border-color)", self.about_css)
        self.assertIn("@media (prefers-reduced-motion: reduce)", self.about_css)

    def test_editorial_chapter_panels_replace_timeline_decoration(self):
        self.assertEqual(self.about_html.count('class="arc-item-label"'), 5)
        self.assertEqual(self.about_html.count('class="arc-index" aria-hidden="true"'), 5)
        self.assertNotIn("timeline-orb", self.about_html)
        self.assertNotIn("timeline-orb", self.about_css)
        self.assertNotIn("<ul", self.about_html)
        self.assertNotIn("<ol", self.about_html)

    def test_chapter_heading_wrap_is_safe(self):
        self.assertIn("overflow-wrap: break-word", self.about_css)
        self.assertNotIn("white-space: nowrap", self.about_css)


if __name__ == "__main__":
    unittest.main()

File: QA/tests/test-about-redesign.js
#!/usr/bin/env node
/**
 * Invariant:
 * - About page chapter structure, chapter navigation, and revised operating-background story must remain intact.
 * Why this exists:
 * - Protects the approved about-page information architecture and content sequencing.
 * What breaks if it fails:
 * - CI blocks deployment to prevent about-page narrative or layout regressions.
 */
const fs = require('fs');
const path = require('path');

const root = path.resolve(__dirname, '..', '..');
const aboutHtml = fs.readFileSync(path.join(root, 'about/index.html'), 'utf8');
const aboutCss = fs.readFileSync(path.join(root, 'styles/about.css'), 'utf8');

const failures = [];

const chapterHeadings = [
  'ENTERPRISE SCALE',
  'GLOBAL DELIVERY LEADERSHIP',
  'GLOBAL INFRASTRUCTURE ADVISORY',
  'AI-ENABLED OPERATING SYSTEMS'
];

for (const heading of chapterHeadings) {
  if (!aboutHtml.includes(heading)) {
    failures.push(`About page missing chapter heading: ${heading}`);
  }
}

if ((aboutHtml.match(/class="era-header"/g) || []).length !== 5) {
failures.push('About page must include exactly five chapter headers.');
}

const requiredSubheadings = [
  'DISNEY &amp; AWS',
  'OMNIGON &amp; ORIGIN DIGITAL',
  'LASER LIGHT COMMUNICATIONS',
  'AGENTIC SOCIETY'
];

for (const subheading of requiredSubheadings) {
  if (!aboutHtml.includes(subheading)) {
    failures.push(`About page missing subheading: ${subheading}`);
  }
}

// Durable, non-time-sensitive construction for the Agentic Society chapter opener.
if (!aboutHtml.includes('Fractional COO work with Agentic Society focuses on')) {
  failures.push('About page must use the approved Fractional COO construction for Agentic Society.');
}

// The substantive claim about a coordinated fleet of agentic AI employees must remain.
if (!aboutHtml.includes('a coordinated fleet of agentic AI employees')) {
  failures.push('About page must preserve the coordinated fleet of agentic AI employees claim.');
}

// Chapter index items must be functional same-page anchors with matching targets.
const chapterAnchorTargets = [
  'enterprise-scale',
  'global-delivery-leadership',
  'global-infrastructure-advisory',
  'ai-enabled-operating-systems',
  'fractional-integration-leadership',
  'operating-philosophy'
];
for (const target of chapterAnchorTargets) {
  if (!aboutHtml.includes(`href="#${target}"`)) {
    failures.push(`About chapter index missing anchor link: #${target}`);
  }
  if (!aboutHtml.includes(`id="${target}"`)) {
    failures.push(`About page missing anchor target: id="${target}"`);
  }
}

const currentToHistoryOrder = [
  'href="#ai-enabled-operating-systems"',
  'href="#fractional-integration-leadership"',
  'href="#global-infrastructure-advisory"',
  'href="#enterprise-scale"',
  'href="#global-delivery-leadership"'
];
let previousOrderIndex = -1;
for (const marker of currentToHistoryOrder) {
  const index = aboutHtml.indexOf(marker);
  if (index <= previousOrderIndex) {
    failures.push('About chapter navigation must run from current responsibility backward through operating history.');
    break;
  }
  previousOrderIndex = index;
}

// Final CTA points to Services with the approved label.
if (!aboutHtml.includes('Explore Service Models') || !aboutHtml.includes('href="/services/"')) {
  failures.push('About final CTA must read Explore Service Models and link to /services/.');
}

// Fixed-header anchor offset must exist for same-page jumps.
if (!aboutCss.includes('scroll-margin-top')) {
  failures.push('About CSS must set scroll-margin-top for fixed-header anchor offset.');
}

// Long chapter labels must be allowed to wrap (no nowrap) so they do not cross the rail.
if (aboutCss.includes('white-space: nowrap')) {
  failures.push('About CSS must not force era headings to a single line (nowrap).');
}

if (aboutHtml.includes('Fractional CEO')) {
  failures.push('About page must not mention Fractional CEO.');
}

if (!aboutHtml.includes('OPERATING PHILOSOPHY')) {
  failures.push('About page must include the approved Operating Philosophy close.');
}

if (aboutHtml.includes('Systems Over Heroics') || aboutHtml.includes('Embedded Operating Leadership')) {
  failures.push('About page must remove the old two-column philosophy card copy.');
}

if (!aboutCss.includes('.about-chapter-nav') || !aboutCss.includes('.arc-item')) {
  failures.push('About CSS must support the chapter nav and editorial chapter panels.');
}

if ((aboutHtml.match(/class="arc-item-label"/g) || []).length !== 5) {
  failures.push('About page must attach a visible label block to each professional-background chapter.');
}

if (aboutHtml.includes('class="arc-index"') || aboutCss.includes('.arc-index')) {
  failures.push('About chapter panels must not render decorative numeric indexes.');
}

if (!aboutHtml.includes('class="advisory-brand"') || !aboutHtml.includes('class="advisory-accent"')) {
  failures.push('About page must visually distinguish Bediner Advisory inside the relationship line.');
}

if (aboutHtml.includes('timeline-orb') || aboutCss.includes('timeline-orb')) {
  failures.push('About page must not retain the obsolete decorative timeline orbs.');
}

if (aboutHtml.includes('<ul') || aboutHtml.includes('<ol')) {
  failures.push('About professional-background narrative must remain prose, not bullets.');
}

if (aboutHtml.includes('class="arc-item"') && !aboutCss.includes('border: 1px solid var(--border-color)')) {
  failures.push('About chapter panels must use the shared border token.');
}

if (!aboutCss.includes('@media (prefers-reduced-motion: reduce)')) {
  failures.push('About chapter motion must include a reduced-motion treatment.');
}

if (!aboutCss.includes('overflow-wrap: break-word')) {
  failures.push('About chapter headings must be allowed to wrap at narrow widths.');
}

if (failures.length > 0) {
  failures.forEach((message) => console.error(`FAIL: ${message}`));
  process.exit(1);
}

console.log('PASS: about redesign and global footer attribution checks passed.');
